webapp/src/ts/modals/edit-report/edit-report.component.ts

```diff
@@ -0,0 +1,81 @@
+import { Component, AfterViewInit } from '@angular/core';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+
+import { MmModalAbstract } from '@mm-modals/mm-modal/mm-modal';
+import { ContactTypesService } from '@mm-services/contact-types.service';
+import { Select2SearchService } from '@mm-services/select2-search.service';
+import { UpdateFacilityService } from '@mm-services/update-facility.service';
+import { take } from 'rxjs/operators';
+
+@Component({
+  selector: 'edit-report',
+  templateUrl: './edit-report.component.html'
+})
+export class EditReportComponent extends MmModalAbstract implements AfterViewInit {
+  static id = 'editReport';
+
+  constructor(
+    bsModalRef: BsModalRef,
+    private contactTypesService:ContactTypesService,
+    private select2SearchService:Select2SearchService,
+    private updateFacilityService:UpdateFacilityService,
+  ) {
+    super(bsModalRef);
+    bsModalRef.onHidden
+      .pipe(take(1)) // so we don't need to unsubscribe
+      // close the select2 popup
+      .subscribe(() => this.getSelectElement().select2('close'));
+  }
+
+  model:any = { report: {} };
```

Comment:
I think this property declaration can be on the top of the class. That keeps them easy to find later

Reply:
I'll totally move it.
As an aside, when I was enabling eslint rules, I was looking over the most common used for `ts` and found one that was called `member-ordering` (https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/rules/member-ordering.ts) which enforces an order on declaring class members. 

I thought it would be super nice to have, but literally NONE of our classes passed so I decided not to add it because of the massive disruption. 

Would you like to have such a rule enabled? cc @craig-landry @garethbowen